app/models/user.rb

```diff
@@ -2,7 +2,10 @@ class User < ApplicationRecord
   has_many :posts, foreign_key: :author_id
   has_many :comments, foreign_key: :author_id
   has_many :likes, foreign_key: :author_id
-  has_many :posts, through: :likes, foreign_key: :author_id
-  has_many :posts, through: :comments, foreign_key: :author_id
+  has_many :liked_posts, through: :likes, source: :post
+  has_many :commented_posts, through: :comments, source: :post
```

Comment:
- [ ] [OPTIONAL]These two are not necessary, these are already achieved by  `has_many :likes & has_many :comments`, you can get all the likes by ex:- `current_user.likes`.